### PR TITLE
Roctracer hip frontend

### PR DESCRIPTION
**Pull request by eshcherb**
ROCM device tracer iplementation

### CODE REVIEW

**Comment by ROCm-Apps-Test** [comment]
Can one of the admins verify this patch?

**Comment by ROCm-Apps-Test** [comment]
Can one of the admins verify this patch?

**Comment by whchung** [comment]
@eshcherb since this PR depends on changes to HIP and HCC to function properly. please help specify them here. also i'd like to ask you to help change the CI script for TensorFlow so this PR can be tested with modified HIP / HCC, as well as ROCm GPU tracer. Please refer to #105 for examples.

File: build_python3 [hunk 1]
```diff
@@ -8,6 +8,9 @@
 #
 # press enter all the way
 #
+# clean the TF cache
+# bazel clean --expunge
```
[the remark this answers is not in the document]
Comment (eshcherb) [reply build_python3]: will remove

File: tensorflow/core/BUILD [hunk 2]
```diff
@@ -2080,7 +2080,8 @@ cc_library(
             "platform/**/env_time.cc",
             "platform/**/cuda_libdevice_path.cc",
             "platform/**/rocm_rocdl_path.cc",
-            "platform/**/device_tracer.cc",
+            "platform/**/device_tracer_cuda.cc",
```
[the remark this answers is not in the document]
Comment (eshcherb) [reply tensorflow/core/BUILD]: The CUDA implementation is all under GOOGLE_CUDA and ROCM implementatino is under TENSORFLOW_USE_ROCM macro.
So they will be excluded by preprocessor, why do I need to check additionally if_cuda_configured in the build script?

File: tensorflow/core/platform/default/device_tracer_cuda.cc [hunk 3]
```diff
@@ -0,0 +1,553 @@
+/* Copyright 2015 The TensorFlow Authors. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (eshcherb) [reply tensorflow/core/platform/default/device_tracer_cuda.cc]: I closed this PR and submitted another one so this is fine in the another PR.

File: tensorflow/core/platform/default/device_tracer_rocm.cc [hunk 4]
```diff
@@ -0,0 +1,553 @@
+/* Copyright 2015 The TensorFlow Authors. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (eshcherb) [reply tensorflow/core/platform/default/device_tracer_rocm.cc]: Will fix

File: tensorflow/core/BUILD (hunk 2 above)
[the remark this answers is not in the document]
Comment (eshcherb) [reply tensorflow/core/BUILD]: The ROCM and CUDA device tracers don't relate to each other, they both implement abstract TC tracer interface.

File: tensorflow/core/platform/default/device_tracer_rocm.cc [hunk 5]
```diff
@@ -0,0 +1,553 @@
...
+  bool Stop() override {
+    mutex_lock l(mu_);
+    if (device_ == NULL) {
+       DT_LOG("DeviceTracerRocm::Stop already disabled");
+       abort();
+    }
+    DT_LOG("DeviceTracerRocm::Stop");
+    device_ = NULL;
+#if 0
+    // Disable HIP API callbacks
```
[the remark this answers is not in the document]
Comment (eshcherb) [reply tensorflow/core/platform/default/device_tracer_rocm.cc]: will fix

File: tensorflow/core/platform/default/device_tracer_rocm.cc [hunk 6]
```diff
@@ -0,0 +1,553 @@
...
+
+  // Activity tracing callback
+  static void activity_callback(const char* begin, const char* end, void* arg) {
+    DeviceTracerRocm* tracer = reinterpret_cast<DeviceTracerRocm*>(arg);
+    DeviceTracerBase* device = tracer->device_;
+    if (device != NULL) {
+      const roctracer_record_t* record = reinterpret_cast<const roctracer_record_t*>(begin);
+      const roctracer_record_t* end_record = reinterpret_cast<const roctracer_record_t*>(end);
+      while (record < end_record) {
+//        const char* name = roctracer_id_string(record->domain, record->activity_kind);
```
[the remark this answers is not in the document]
Comment (eshcherb) [reply tensorflow/core/platform/default/device_tracer_rocm.cc]: sure

**Pull request closed without merging**